### PR TITLE
Fix cache key for files due to missing log history

### PR DESCRIPTION
**Pull request by eukarpov**

### CODE REVIEW

**Comment by Blackhex** [comment]
Wouldn't be better to compute hash of the affected files content completely without involving git commit hashes? This would allow cache hit betwen in progress PR commit and the master build after the PR merge as well.

**Comment by eukarpov** [comment]
> Wouldn't be better to compute hash of the affected files content completely without involving git commit hashes? This would allow cache hit betwen in progress PR commit and the master build after the PR merge as well. Ofc, this would require checking out the repo first.

It is also an option, however it is more complicated to calculate especially for folders

**Comment by Blackhex** [comment]
Alternatively https://git-scm.com/docs/git-hash-object might be an option. I am not sure whether this hash is stable accross commits though.

**Comment by eukarpov** [comment]
I suggest to start with this solution because it is working and already helping to build FFmpeg. Then it can be improved if needed

**Comment by Blackhex** [comment]
I suggest to use:

```
gh api repos/Windows-on-ARM-Experiments/mingw-woarm64-build/contents/.github/workflows/advanced.yml -q .sha
```

for single files and 

```
gh api repos/Windows-on-ARM-Experiments/mingw-woarm64-build/contents/.github/workflows --paginate | jq
 -r '.[] | .name + " " + .sha'
```

for directories.

File: .github/workflows/advanced.yml [hunk 1]
```diff
@@ -72,6 +72,9 @@ env:
   SOURCE_PATH: ${{ github.workspace }}/code
   ARTIFACT_PATH: ${{ github.workspace }}/artifact
 
+  GITHUB_EVENT_PULL_REQUEST_BASE_REF: ${{ github.event.pull_request.base.ref }}
```
Comment (Blackhex) [inline .github/workflows/advanced.yml:75]: I'd call it just `BASE_REF`  and `HEAD_REF` the `GITHUB_EVENT_PULL_REQUEST_` prefix is just a noise.

File: .github/scripts/get-files-sha.sh [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+
+    if ! git rev-parse --verify "$GITHUB_EVENT_PULL_REQUEST_HEAD_REF"; then
+        git fetch origin $GITHUB_EVENT_PULL_REQUEST_HEAD_REF
+        git branch $GITHUB_EVENT_PULL_REQUEST_HEAD_REF FETCH_HEAD
+    fi
+
+    sha=$(git log -1 --format="%H" origin/$GITHUB_EVENT_PULL_REQUEST_BASE_REF $path)
+    sha=$sha-$(git log -1 --format="%H" origin/$GITHUB_EVENT_PULL_REQUEST_HEAD_REF $path)
+    echo $sha
+    echo sha=$sha >> "$GITHUB_OUTPUT"
```
Comment (Blackhex) [inline .github/scripts/get-files-sha.sh:21]: When `GITHUB_EVENT_PULL_REQUEST_BASE_REF` and  `GITHUB_EVENT_PULL_REQUEST_HEAD_REF` will have shorter name, the lines 18-21 can be just single line command.

**Review by Blackhex — APPROVED** [review]
I can approve as a temporary solution though this approach does not work 100% correctly in a sense of effective caching.

File: .github/workflows/advanced.yml (hunk 1 above)
Comment (eukarpov) [reply .github/workflows/advanced.yml]: Done

File: .github/scripts/get-files-sha.sh (hunk 2 above)
Comment (Blackhex) [reply .github/scripts/get-files-sha.sh:21]: The suggestion was:
```
echo sha=$(git log -1 --format="%H" origin/$BASE_REF $path)-$sha-$(git log -1 --format="%H" origin/$HEAD_REF $path) >> "$GITHUB_OUTPUT"
```

Comment (eukarpov) [reply .github/scripts/get-files-sha.sh]: Not needed anymore

File: .github/scripts/get-files-sha.sh [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+    exit 0
```
[the remark this answers is not in the document]
Comment (eukarpov) [reply .github/scripts/get-files-sha.sh]: Not needed anymore

**Comment by eukarpov** [comment]
It looks like hash is stable on every merge, original assumption was wrong due to missing log history

**Pull request merged**